### PR TITLE
Updated changelog to reflect changes and added more metrics to the me…

### PR DESCRIPTION
**Pull request by RycroftSolutions**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass 

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Known Compatablity Issues

…trics-mongodb.rb script

### CODE REVIEW

**Comment by RycroftSolutions** [comment]
Added more metrics to the metrics-mongodb.rb

**Comment by sstarcher** [comment]
The rubocop error is due to the excessive assignments in the function.  Either split those into separate functions or tell rubocop to ignore that specific rubocop check

```
bin/metrics-mongodb.rb:158:3: C: Assignment Branch Condition size for gather_replication_metrics is too high. [397.2/100]
  def gather_replication_metrics(server_status)
  ^^^
```

**Comment by RycroftSolutions** [comment]
I think all issues are now resolved.

Please confirm if this pull request can be merged or if I need to create a new pull request

**Comment by RycroftSolutions** [comment]
Thanks again for all your time helping me through this. I am officially a stage 2 newbie now :)

**Comment by sstarcher** [comment]
@RycroftSolutions no problem everyone is at one point.  Thanks for the contribution.

**Comment by RycroftSolutions** [comment]
Hi Shane,

Is there an additional process I have not completed?

The reason I ask is because I just tried to pull down the metrics-mongodb.rb script using the “sensu-install -p mongodb” and it seems I can only get release 1.0.0 which doesn’t carry my changes.

Any advice is appreciated.

Thanks,
Neil

**Comment by sstarcher** [comment]
@RycroftSolutions a new release has not be cut yet.  Someone will have to cut a release for you @sensu-plugins/commit-bit

**Comment by RycroftSolutions** [comment]
Do you know how I can get the ball rolling to get this done?

**Comment by sstarcher** [comment]
I pinged the maintainers in the last message.  I'm not certain on the proper mechanism.  The proper way may be to open a ticket asking for a release.

**Comment by eheydrick** [comment]
New release coming soon.

File: bin/metrics-mongodb.rb [hunk 1]
```diff
@@ -155,41 +155,165 @@ def run
...
+
+    # Extra info
+    server_metrics['mem.heap_usage_bytes'] = server_status['extra_info']['heap_usage_bytes']
+    server_metrics['mem.pageFaults'] = server_status['extra_info']['page_faults']
+
+    # Global Lock
+    server_metrics['lock.totalTime'] = server_status['globalLock']['totalTime']
+    server_metrics['lock.queue_total'] = server_status['globalLock']['currentQueue']['total']
+    server_metrics['lock.queue_readers'] = server_status['globalLock']['currentQueue']['readers']
+    server_metrics['lock.queue_writers'] = server_status['globalLock']['currentQueue']['writers']
```
Comment (eheydrick) [inline bin/metrics-mongodb.rb]: these lock queue metrics were previously `lock.queue.thing`. can you change it back to that so we maintain backwards compatibility.

**Comment by eheydrick** [comment]
Hey @RycroftSolutions I dig the additional metrics. Can you address my comment about backwards compatibility on the lock queue metrics. Once that's resolved we'll cut a new release.

**Comment by RycroftSolutions** [comment]
Hi Eric,

Thanks for the feedback I have updated the lock.queue statements and created a pull request if you can do the honours.

Best regards,
Neil

**Comment by RycroftSolutions** [comment]
Hi Eric, I have tried everything and seem to have ended up with a pull request so please let me know if I should just create a new branch and start from scratch again

**Pull request merged**